src/test/kotlin/com/github/jensim/megamanipulator/actions/git/commit/CommitOperatorTest.kt

```diff
@@ -0,0 +1,270 @@
+package com.github.jensim.megamanipulator.actions.git.commit
+
+import com.github.jensim.megamanipulator.actions.ProcessOperator
+import com.github.jensim.megamanipulator.actions.apply.ApplyOutput
+import com.github.jensim.megamanipulator.actions.localrepo.LocalRepoOperator
+import com.github.jensim.megamanipulator.actions.search.SearchResult
+import com.github.jensim.megamanipulator.actions.vcs.PrRouter
+import com.github.jensim.megamanipulator.settings.CodeHostSettings
+import com.github.jensim.megamanipulator.settings.ForkSetting
+import com.github.jensim.megamanipulator.settings.MegaManipulatorSettings
+import com.github.jensim.megamanipulator.settings.SettingsFileOperator
+import com.github.jensim.megamanipulator.ui.DialogGenerator
+import com.github.jensim.megamanipulator.ui.UiProtector
+import com.intellij.openapi.project.Project
+import io.mockk.Called
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.every
+import io.mockk.impl.annotations.InjectMockKs
+import io.mockk.impl.annotations.MockK
+import io.mockk.junit5.MockKExtension
+import io.mockk.mockk
+import io.mockk.slot
+import io.mockk.verify
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.runBlocking
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.equalTo
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.NullAndEmptySource
+import org.junit.jupiter.params.provider.ValueSource
+import java.io.File
+import java.nio.file.Path
+import java.util.concurrent.ConcurrentHashMap
+import kotlin.io.path.ExperimentalPathApi
+import kotlin.io.path.createTempDirectory
+import kotlin.test.assertTrue
+
+@ExperimentalPathApi
+@ExtendWith(MockKExtension::class)
+class CommitOperatorTest {
+
+    @MockK
+    private lateinit var dialogGenerator: DialogGenerator
+
+    @MockK
+    private lateinit var settingsFileOperator: SettingsFileOperator
+
+    @MockK
+    private lateinit var localRepoOperator: LocalRepoOperator
+
+    @MockK
+    private lateinit var processOperator: ProcessOperator
+
+    @MockK
+    private lateinit var prRouter: PrRouter
+
+    @MockK(relaxed = true)
+    private lateinit var uiProtector: UiProtector
```

Comment:
Dont mock this, use the TestUiProtector for direct execution

Reply:
ok maybe ill have to rewrite the tests now, it will take sometime